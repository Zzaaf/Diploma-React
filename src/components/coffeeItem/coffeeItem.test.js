import React from 'react';
import {shallow} from 'enzyme';
import CoffeeService from '../../services/coffeeService';
import CoffeeItem from '../coffeeItem';
import ErrorMessage from '../errorMessage';

describe('Testing <CoffeeItem/>', () => {
    const service = new CoffeeService();
    const item = shallow(<CoffeeItem
            getData={service.getAllCoffee}
            />);
    describe('Testing snap & state', () => {
        it('CoffeeItem has rendered correctly', () => {
            expect(item).toMatchSnapshot();
        });
        it('CoffeeItem state "posts" is array', () => {
            expect(item.state().post).toBeArray();
        });
        it('CoffeeItem state "loading" is true', () => {
            expect(item.state().loading).toBeTruthy();
        });
        it('CoffeeItem state "error" is false', () => {
            expect(item.state().error).toBeFalsy();
        });
        it('CoffeeItem state "typeError" is string', () => {
            expect(item.state().typeError).toBeString();
        });
        it('CoffeeItem state "fatalError" is false', () => {
            expect(item.state().fatalError).toBeFalsy();
        });
    });
    describe('Handlers tests', () => {
        it('Testing onError', () => {
            item.instance().onError(<ErrorMessage typeError='404'/>);
            expect(item.state().error).toBeTruthy();
            expect(item.state().loading).toBeFalsy();
        });
    });
});