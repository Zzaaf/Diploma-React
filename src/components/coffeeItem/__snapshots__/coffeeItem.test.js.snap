// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CoffeeItem/> Testing snap & state CoffeeItem has rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CoffeeItem
    getData={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Container
        tag="div"
      >
        <Row
          tag="div"
        >
          <Spinner />
        </Row>
      </Container>,
      "className": "shop",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Row
          tag="div"
        >
          <Spinner />
        </Row>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            null,
            <Spinner />,
            null,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          null,
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Container
          tag="div"
        >
          <Row
            tag="div"
          >
            <Spinner />
          </Row>
        </Container>,
        "className": "shop",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Row
            tag="div"
          >
            <Spinner />
          </Row>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              null,
              <Spinner />,
              null,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            null,
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
